lib/actions/api.js

```diff
@@ -45,22 +49,75 @@ export function planTrip (customOtpQueryBuilder) {
   }
 }
 
+// Profile Query
+
+export const receivedProfileError = createAction('PROFILE_ERROR')
+export const receivedProfileResponse = createAction('PROFILE_RESPONSE')
+export const requestProfileResponse = createAction('PROFILE_REQUEST')
+
+function constructPlanQuery (api, query) {
+  const planEndpoint = `${api.host}${api.port ? ':' + api.port : ''}${api.path}/plan`
+
+  let params = {}
+  queryParams.filter(qp => qp.planTypes.indexOf('ITINERARY') !== -1 && qp.name in query).forEach(qp => {
```

Comment:
I think it would be good to have this filtering out of values also print warnings when an unrecognized key is sent.